File: kattis/foundation/quadrant/quadrant.py
import sys
import time
from typing import Optional

sys.setrecursionlimit(300000)

class FastScanner:
    tokens = None

    @classmethod
    def init(cls) -> None:
        cls.tokens = iter(sys.stdin.read().split())

    @classmethod
    def next_string(cls) -> Optional[str]:
        try:
            return next(cls.tokens)
        except StopIteration:
            return None

    @classmethod
    def next_int(cls) -> Optional[int]:
        token = cls.next_string()
        return int(token) if token is not None else None

    @classmethod
    def next_double(cls) -> Optional[float]:
        token = cls.next_string()
        return float(token) if token is not None else None

class quadrant:
    DEBUG = False
    _start_runtime = 0.0

    @classmethod
    def init(cls) -> None:
        FastScanner.init()
        if cls.DEBUG:
            cls._start_runtime = time.perf_counter()

    @classmethod
    def close(cls) -> None:
        """Evaluates hardware runtime execution speed."""
        if cls.DEBUG:
            total_runtime = time.perf_counter() - cls._start_runtime
            print(f"[Algorithmic Runtime] {total_runtime:.6f} seconds", file=sys.stderr)

    @classmethod
    def d_print(cls, *args, sep=" ") -> None:
        """Enterprise trace logger. Always routes to stderr safely."""
        if cls.DEBUG:
            print(f"[DEBUG] {sep.join(map(str, args))}", file=sys.stderr)
    
    @classmethod  
    def isValidCoordinate(cls, value: int) -> bool:
        min = -1000
        max = 1000
        return min <= value <= max and value != 0

    @classmethod
    def solve(cls) -> None:
        # First loading
        next_string = FastScanner.next_string
        next_int = FastScanner.next_int
        write = sys.stdout.write
        d_print = cls.d_print

        # Take the coordinate input
        inputX = next_int()
        inputY = next_int()
        extraInput = next_string()
        if inputX is None or inputY is None or extraInput is not None:
            d_print("Wrong input")
            return

        # Is valid input
        d_print("Parsed Coordinates -> X:", inputX, "Y:", inputY)
        if any(not cls.isValidCoordinate(coordinate) for coordinate in [inputX, inputY]):
            d_print("Invalid input")

        # Find quadrant
        if inputX > 0 and inputY > 0:
            write("1\n")
        elif inputX < 0 and inputY > 0:
            write("2\n")
        elif inputX < 0 and inputY < 0:
            write("3\n")
        elif inputX > 0 and inputY < 0:
            write("4\n")
        else:
            d_print("Unknown coordinate")


if __name__ == "__main__":
    quadrant.init()
    quadrant.solve()
    quadrant.close()

File: kattis/foundation/quadrant/quadrant.java
package kattis.foundation.quadrant;

import java.io.InputStream;
import java.io.PrintWriter;
import java.io.BufferedOutputStream;

public class quadrant {

    private static final boolean DEBUG = false;
    private static PrintWriter out;
    private static long startRuntime;

    private static void init() {
        out = new PrintWriter(new BufferedOutputStream(System.out));

        if (DEBUG) {
            startRuntime = System.nanoTime();
        }
    }

    private static void close() {
        if (DEBUG) {
            long endRuntime = System.nanoTime();
            double totalRuntime = (endRuntime - startRuntime) / 1_000_000_000.0;
            System.err.println(String.format("[Algorithmic Runtime] %.6f seconds", totalRuntime));
        }

        out.flush();
    }

    private static void solve() {
        // Take the coordinate input
        Integer inputX = FastScanner.nextInt();
        Integer inputY = FastScanner.nextInt();
        boolean extraInput = FastScanner.hasNext();
        if (inputX == null || inputY == null || extraInput) {
            dPrint("Wrong input");
            return;
        }

        // Is valid input
        dPrint("Parsed Coordinates -> X:", inputX, "Y:", inputY);
        if (!isValidCoordinate(inputX) || !isValidCoordinate(inputY)) {
            dPrint("Invalid input");
            return;
        }

        // Find quadrant
        if (inputX > 0 && inputY > 0) {
            out.println("1");
        } else if (inputX < 0 && inputY > 0) {
            out.println("2");
        } else if (inputX < 0 && inputY < 0) {
            out.println("3");
        } else if (inputX > 0 && inputY < 0) {
            out.println("4");
        } else {
            out.println("Unknown coordinate");
        }
    }

    public static void main(String[] args) {
        init();
        solve();
        close();
    }

    static class FastScanner {
        private static final int BUFFER_SIZE = 1 << 16; // 64KB hardware chunk block
        private static final InputStream in = System.in;
        private static final byte[] buffer = new byte[BUFFER_SIZE];
        private static int bufferPointer = 0;
        private static int bytesRead = 0;

        private static boolean fillBuffer() {
            try {
                bytesRead = in.read(buffer, 0, BUFFER_SIZE);
                bufferPointer = 0;
                return bytesRead > 0;
            } catch (Exception e) {
                return false;
            }
        }

        private static byte readByte() {
            if (bufferPointer >= bytesRead) {
                if (!fillBuffer())
                    return -1;
            }
            return buffer[bufferPointer++];
        }

        public static boolean hasNext() {
            while (true) {
                if (bufferPointer >= bytesRead) {
                    if (!fillBuffer())
                        return false;
                }
                if (buffer[bufferPointer] > 32)
                    return true;
                bufferPointer++;
            }
        }

        public static int nextInt() {
            byte c = readByte();
            while (c <= 32) {
                if (c == -1)
                    return 0;
                c = readByte();
            }
            boolean negative = false;
            if (c == '-') {
                negative = true;
                c = readByte();
            }
            int value = 0;
            while (c > 32) {
                if (c >= '0' && c <= '9') {
                    value = (value * 10) + (c - '0');
                }
                c = readByte();
            }
            return negative ? -value : value;
        }

        public static long nextLong() {
            byte c = readByte();
            while (c <= 32) {
                if (c == -1)
                    return 0L;
                c = readByte();
            }
            boolean negative = false;
            if (c == '-') {
                negative = true;
                c = readByte();
            }
            long value = 0;
            while (c > 32) {
                if (c >= '0' && c <= '9') {
                    value = (value * 10) + (c - '0');
                }
                c = readByte();
            }
            return negative ? -value : value;
        }

        public static double nextDouble() {
            if (!hasNext())
                return 0.0;
            byte c = readByte();
            while (c <= 32)
                c = readByte();
            boolean negative = false;
            if (c == '-') {
                negative = true;
                c = readByte();
            }
            double value = 0;
            while (c > 32 && c != '.') {
                if (c >= '0' && c <= '9') {
                    value = (value * 10) + (c - '0');
                }
                c = readByte();
            }
            if (c == '.') {
                c = readByte();
                double multiplier = 1.0;
                while (c > 32) {
                    if (c >= '0' && c <= '9') {
                        multiplier /= 10.0;
                        value += (c - '0') * multiplier;
                    }
                    c = readByte();
                }
            }
            return negative ? -value : value;
        }

        public static String next() {
            if (!hasNext())
                return null;
            StringBuilder sb = new StringBuilder();
            byte c = readByte();
            while (c <= 32) {
                if (c == -1)
                    return null;
                c = readByte();
            }
            while (c > 32) {
                sb.append((char) c);
                c = readByte();
            }
            return sb.toString();
        }
    }

    private static void dPrint(Object message) {
        if (DEBUG)
            System.err.println("[DEBUG] " + message);
    }

    private static void dPrint(Object... args) {
        if (DEBUG) {
            StringBuilder sb = new StringBuilder("[DEBUG] ");
            for (int i = 0; i < args.length; i++) {
                sb.append(args[i]).append(i < args.length - 1 ? " " : "");
            }
            System.err.println(sb.toString());
        }
    }

    private static boolean isValidCoordinate(int value) {
        int min = -1000;
        int max = 1000;
        return min <= value && value <= max && value != 0;
    }
}